### PR TITLE
refactor for character references to accomodate adding in alerts

### PR DESCRIPTION
**Pull request by SoLetsDev**
ECER-{933}: Refactoring character references to include alerts.
## Checklist
- [x] I have tested these changes locally.
- [ ] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

Object will get stored like this in our WizardData 
image

What it looks like 
image

Character reference preview page
image

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue [hunk 1]
```diff
@@ -0,0 +1,119 @@
+<template>
+  <v-row>
+    <v-col>
+      <Alert type="info">
+        <p>The Registry will not assess an application until references have been submitted. Please make sure your reference:</p>
+        <ul class="pl-5">
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue:6]: this was needed due to a bug with vuetify and ol/ul not showing their bullets. 

https://github.com/vuetifyjs/vuetify/issues/17633

**Review by farzadnadiri — DISMISSED** [review]

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/utils/formRules.ts [hunk 2]
```diff
@@ -87,7 +87,13 @@ const hasCheckbox = (message = "You must check the box") => {
...
 const required = (message = "This field is required") => {
-  return (v: string) => !!(v && v.trim()) || message;
+  return (v: string | number) => {
+    if (typeof v === "number") {
+      return !!v || message;
+    } else {
+      return !!(v && v?.trim()) || message;
+    }
+  };
 };
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/utils/formRules.ts:97]: Thank you 😎

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue (hunk 1 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue:6]: This has been annoying me also

**Review by pcatkins — APPROVED** [review]

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/utils/formRules.ts (hunk 2 above)
Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/utils/formRules.ts:97]: np :)

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue (hunk 1 above)
Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReference.vue:6]: Yeah pl-5 haha quick hack for now :P

**Pull request merged**